tier1/convar.cpp

```diff
@@ -552,582 +456,56 @@ void ConCommand::Init()
-// Purpose: 
-// Input  : value - 
-//-----------------------------------------------------------------------------
-void ConVar::SetValue( float value )
-{
-	ConVar *var = ( ConVar * )m_pParent;
-	var->InternalSetFloatValue( value );
-}
-
-//-----------------------------------------------------------------------------
-// Purpose: 
-// Input  : value - 
-//-----------------------------------------------------------------------------
-void ConVar::SetValue( int value )
-{
-	ConVar *var = ( ConVar * )m_pParent;
-	var->InternalSetIntValue( value );
-}
-
-//-----------------------------------------------------------------------------
-// Purpose: 
-// Input  : value - 
-//-----------------------------------------------------------------------------
-void ConVar::SetValue( Color value )
-{
-	ConVar *var = ( ConVar * )m_pParent;
-	var->InternalSetColorValue( value );
-}
-
-//-----------------------------------------------------------------------------
-// Purpose: Reset to default value
-//-----------------------------------------------------------------------------
-void ConVar::Revert( void )
-{
-	// Force default value again
-	ConVar *var = ( ConVar * )m_pParent;
-	var->SetValue( var->m_pszDefaultValue );
-}
-
-//-----------------------------------------------------------------------------
-// Purpose: 
-// Input  : minVal - 
-// Output : true if there is a min set
-//-----------------------------------------------------------------------------
-bool ConVar::GetMin( float& minVal ) const
-{
-	minVal = m_pParent->m_fMinVal;
-	return m_pParent->m_bHasMin;
-}
-
-//-----------------------------------------------------------------------------
-// Purpose: 
-// Input  : maxVal - 
-//-----------------------------------------------------------------------------
-bool ConVar::GetMax( float& maxVal ) const
-{
-	maxVal = m_pParent->m_fMaxVal;
-	return m_pParent->m_bHasMax;
-}
-
-//-----------------------------------------------------------------------------
-// Purpose: 
-// Output : const char
-//-----------------------------------------------------------------------------
-const char *ConVar::GetDefault( void ) const
-{
-	return m_pParent->m_pszDefaultValue;
-}
-
-
-//-----------------------------------------------------------------------------
-// This version is simply used to make reading convars simpler.
-// Writing convars isn't allowed in this mode
-//-----------------------------------------------------------------------------
-class CEmptyConVar : public ConVar
-{
-public:
-	CEmptyConVar() : ConVar( "", "0" ) {}
-	// Used for optimal read access
-	virtual void SetValue( const char *pValue ) {}
-	virtual void SetValue( float flValue ) {}
-	virtual void SetValue( int nValue ) {}
-	virtual void SetValue( Color cValue ) {}
-	virtual const char *GetName( void ) const { return ""; }
-	virtual bool IsFlagSet( int nFlags ) const { return false; }
+void* invalid_convar[EConVarType_MAX + 1] = {

```

Comment:
Potentially move that to a static initialization inside the GetInvalidConVar() function or somewhere alike, to not pollute the global scope.